main/testkit/src/mill/testkit/MillTestkit.scala

```diff
@@ -20,8 +20,6 @@ trait MillTestKit {
 
   def targetDir: os.Path = defaultTargetDir
 
-  def externalOutPath: os.Path = targetDir / "external"
```

Comment:
This is an bug fix in the test harness; previously, external modules like `mill.scalalib.ZincWorkerModule` were not getting their external module folders deleted between runs, resulting in weird interference between test runs that are meant to be independent. I tweaked the `TestEvaluator` use the same `out/` folder as it uses for normal modules, which is also the setup we use in production. That makes the bug go away and also brings the test and prod configurations closer together